tests/Functional/lib/AutoregisterClassmapPluginTest.php

```diff
@@ -0,0 +1,202 @@
+     */
+    const AUTOLOAD_GENERATED_DIR = './tmp/composer/';
+
+    /**
+     * @var string AUTOLOAD_FILENAME
+     */
+    const AUTOLOAD_FILENAME = 'autoload_psr4.php';
+
+    /**
+     * @var AutoregisterClassmapPlugin
+     */
+    protected $autoregister;
+
+    /**
+     * @var Event
+     */
+    protected $event;
+
+    /**
+     * @var Composer|Phake_IMock
+     */
+    protected $composer;
+
+    /**
+     * @var IOInterface
+     */
+    protected $interface;
+
+    /**
+     * @var RootPackage|Phake_IMock
+     */
+    protected $package;
+
+    /**
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * @var Filesystem
+     */
+    protected $fs;
+
+    /**
+     * @var Finder
+     */
+    protected $finder;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setUp()
+    {
+        $this->event        = Phake::mock(Event::class);
+        $this->composer     = Phake::mock(Composer::class);
+        $this->interface    = Phake::mock(IOInterface::class);
+        $this->package      = Phake::mock(RootPackage::class);
+        $this->config       = Phake::mock(Config::class);
+        $this->fs           = new Filesystem();
+        $this->finder       = new Finder();
+        $this->autoregister = new AutoregisterClassmapPlugin();
+    }
+
+    /**
+     * Set up to define the return on methods called
+     */
+    public function setUpMethodCalled()
+    {
+        Phake::when($this->composer)->getPackage()->thenReturn($this->package);
+        Phake::when($this->package)->getExtra()->thenReturn($this->getExtraConfig());
+        Phake::when($this->composer)->getConfig()->thenReturn($this->config);
+        Phake::when($this->config)->get('vendor-dir')->thenReturn(self::VENDOR_DIR);
+    }
+
+    /**
+     * Test the autoload files generation
+     *
+     * @dataProvider listFilesGenerated
+     */
+    public function testGenerateAutoloadFiles($fileExpected)
+    {
+        $this->clean();
+        $this->setUpMethodCalled();
+
+        $this->autoregister->activate($this->composer, $this->interface);
+
+        $this->autoregister->moveFile($this->event);
+        $this->autoregister->run($this->event);
+
+        $this->assertTrue($this->fs->exists(static::VENDOR_DIR . DIRECTORY_SEPARATOR . $fileExpected));
+    }
+
+    /**
+     * Test contents in autoload generated
+     */
+    public function testContentsInAutoloadGenerated()
+    {
+        $this->clean();
+        $this->setUpMethodCalled();
+
+        $this->autoregister->activate($this->composer, $this->interface);
+
+        $this->autoregister->moveFile($this->event);
+        $this->autoregister->run($this->event);
+
+        $this->finder->files()->in(static::VENDOR_DIR);
+
+        /** @var SplFileInfo $file */
+        foreach ($this->finder as $file) {
+            if ('autoload_psr4.php' === $file->getFilename()) {
+                $this->assertContains('AppModule\\\\Module\\\\Bundle\\\\TestBundle\\\\', $file->getContents());
```

Comment:
In your file generated you should tag it with "\@generated by autoregister plugin"